sdk/tables/azure-data-tables/azure/data/tables/_generated/_configuration.py

```diff
@@ -6,54 +6,56 @@
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
 
-from typing import TYPE_CHECKING
+import sys
+from typing import Any
 
 from azure.core.configuration import Configuration
 from azure.core.pipeline import policies
 
 from ._version import VERSION
 
-if TYPE_CHECKING:
-    # pylint: disable=unused-import,ungrouped-imports
-    from typing import Any
+if sys.version_info >= (3, 8):
+    from typing import Literal  # pylint: disable=no-name-in-module, ungrouped-imports
+else:
+    from typing_extensions import Literal  # type: ignore  # pylint: disable=ungrouped-imports
 
 
-class AzureTableConfiguration(Configuration):
+class AzureTableConfiguration(Configuration):  # pylint: disable=too-many-instance-attributes
     """Configuration for AzureTable.
 
     Note that all parameters used to create this instance are saved as instance
     attributes.
 
-    :param url: The URL of the service account or table that is the targe of the desired operation.
+    :param url: The URL of the service account or table that is the target of the desired
+     operation. Required.
     :type url: str
+    :keyword version: Specifies the version of the operation to use for this request. Default value
+     is "2019-02-02". Note that overriding this default value may result in unsupported behavior.
+    :paramtype version: str
     """
 
-    def __init__(
-        self,
-        url,  # type: str
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> None
+    def __init__(self, url: str, **kwargs: Any) -> None:
+        super(AzureTableConfiguration, self).__init__(**kwargs)
+        version = kwargs.pop("version", "2019-02-02")  # type: Literal["2019-02-02"]
+
         if url is None:
             raise ValueError("Parameter 'url' must not be None.")
-        super(AzureTableConfiguration, self).__init__(**kwargs)
 
         self.url = url
-        self.version = "2019-02-02"
-        kwargs.setdefault('sdk_moniker', 'data-tables/{}'.format(VERSION))
+        self.version = version
+        kwargs.setdefault("sdk_moniker", "table/{}".format(VERSION))
```

Comment:
Looks like me might need to correct the user-agent format.

Reply:
Looks like this is being set elsewhere:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/azure/data/tables/_base_client.py#L239
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/azure/data/tables/_sdk_moniker.py#L11